File: Data Structures & Algorithms/partition-labels/submission-3.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
        int n = s.size();
        unordered_map<char, int> lastIndMap;
        for(int i = 0; i < n; i++) {
            lastIndMap[s[i]] = max(i, lastIndMap[s[i]]);
        }

        int l = 0;
        int temp = -1;
        vector<int> res;
        for(int r = 0; r < n; r++) {
            temp = max(temp, lastIndMap[s[r]]);
            if(temp == r){
                res.push_back(r - l + 1);
                l = r + 1;
            }
        }
        return res;
        
    }
};

File: Data Structures & Algorithms/palindrome-number/submission-3.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        d = 1
        while x // (d * 10) != 0:
            d *= 10
        
        while x:
            l = x // d
            r = x % 10
            if l != r:
                return False
            x = x % d
            x = x // 10
            d = d // 100
        
        return True